src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -696,6 +755,24 @@ public void deleteDBs() throws Exception {
     }
   }
 
+  /**
+   * This is called by operations outside of the ReaderMetricsProcessor.
+   *
+   * @return A list of the timestamps associated with MetricsDB files. The oldest of the MetricsDB files typically
+   *     have a lifetime of ~SAMPLING_INTERVAL seconds (no less than SAMPLING_INTERVAL/2 seconds). Null if batch
+   *     metrics is disabled.
+   */
+  public NavigableSet<Long> getBatchMetrics() {
+    if (batchMetricsEnabled) {
+      TreeSet<Long> batchMetricsDBSetCopy = new TreeSet<>(batchMetricsDBSet.clone());
+      while (batchMetricsDBSetCopy.size() > PluginSettings.instance().getBatchMetricsRetentionPeriodMinutes() * 12) {
```

Comment:
`PluginSettings.instance().getBatchMetricsRetentionPeriodMinutes() * 12`

This is a call we are making on each iteration of the loop. Can we store it in a local variable as it does not change over the loop iterations ?

Reply:
Sure